test/unit/test-amp-img.js

```diff
@@ -406,7 +407,10 @@ describes.sandboxed('amp-img', {}, (env) => {
     el.getLayoutSize = () => ({width: 300, height: 200});
 
     const impl = new AmpImg(el);
-    const propagateAttributesSpy = sandbox.spy(impl, 'propagateAttributes');
+    const propagateAttributesSpy = sandbox.spy(
```

Comment:
I know this is how the code was beforehand, but it seems like instead of testing that this helper is called we should directly test that the expected attributes were propagated.

Reply:
That's a really good point. I'll switch this over.

Reply:
Unfortunately this test ensures the order of attributes being propagated, and ensures `src` is last or double downloads can occur in Webkit.

I'll see if I can come up with a different method of verification.

Reply:
Perhaps instead spy on the element itself a la something like `env.spy(impl.element, 'setAttribute')`?

Reply:
That is better than the change I made (validated order of attributes to propagate).